doc/manual/source/user_guide/InSituPythonAnalysis.rst

```diff
@@ -0,0 +1,269 @@
+      // Put yt_run_Function and yt_run_FunctionArguments here
+
+  inside ``CallInSitulibyt`` function in ``src/enzo/CallInSitulibyt.C`` according to our needs.
+
+  Please make sure the functions we called are defined inside the script. Otherwise, in ``libyt`` normal modes, the simulation will terminate simply because it cannot find the Python function, while in the other modes, it will labeled as failed.
+
+  See how to use yt to do analysis `here <https://libyt.readthedocs.io/en/latest/in-situ-python-analysis/using-yt.html>`__.
+
+.. _Interactive Python Prompt:
+
+Interactive Python Prompt
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **How to activate interactive Python prompt in Enzo?**
+
+  We have to compile ``libyt`` in **interactive mode** and then un-comment the code in ``src/enzo/CallInSitulibyt.C``:
+
+  ::
+
+    /* Call interactive Python prompt. */
+    if (yt_run_InteractiveMode("LIBYT_STOP") != YT_SUCCESS) {
+        fprintf(stderr, "Error in libyt API yt_run_InteractiveMode\n");
+        fprintf(stderr, "One reason might be compiling libyt without -DINTERACTIVE_MODE=ON, "
+                        "which does not support yt_run_InteractiveMode.\n");
+    }
+
+  If Enzo detects ``LIBYT_STOP`` file, then interactive Python prompt will activate.
+  We can find more about libyt API ``yt_run_InteractiveMode`` `here <https://libyt.readthedocs.io/en/latest/libyt-api/yt_run_interactivemode.html>`__.
+
+* **How to use interactive Python prompt? How does it work?**
+
+  It is like a normal Python prompt but with access to simulation data,
+  see `here <https://libyt.readthedocs.io/en/latest/in-situ-python-analysis/interactive-python-prompt.html>`__ for how to use interactive Python prompt.
+
+  Interactive Python prompt only works on local desktops or submit an interactive job to HPC cluster (ex: ``qsub -I`` in PBS scheduler),
+  because the prompt gets inputs from the terminal.
+  The root process gets the inputs and then broadcasts the inputs to every MPI process. They run the statements synchronously.
+
+.. _Reloading Script:
+
+Reloading Script
+^^^^^^^^^^^^^^^^
+
+* **How to activate reload Python script in Enzo?**
+
+  We have to compile ``libyt`` in **interactive mode** and then un-comment the code in ``src/enzo/CallInSitulibyt.C``:
+
+  ::
+
+    /* Reloading script */
+    // if (yt_run_ReloadScript("LIBYT_STOP", "RELOAD", "reload.py") != YT_SUCCESS) {
+    //     fprintf(stderr, "Error in libyt API yt_run_ReloadScript\n");
+    //     fprintf(stderr, "One reason might be compiling libyt without -DINTERACTIVE_MODE=ON, "
+    //                     "which does not support yt_run_ReloadScript.\n");
+    // }
+
+  If an error occurred when running inline Python functions or Enzo detects ``LIBYT_STOP`` file, then it will enter reloading script phase.
+  Document about ``yt_run_ReloadScript`` is `here <https://libyt.readthedocs.io/en/latest/libyt-api/yt_run_reloadscript.html>`__.
+
+* **How to reload a script?**
```

Comment:
It's not clear to me why I would want to use this feature; an example use case would really help. Even just a few sentences.

Reply:
This is fixed.